cloud/src/main/java/org/jimmutable/cloud/elasticsearch/ElasticSearchRESTClient.java

```diff
@@ -1064,17 +1079,20 @@ public boolean writeAllToCSV( IndexDefinition index, String query_string, List<S
 						return false;
 					}
 				}
-				ScrollResponse<Map> scroll_response = esClient.scroll(new ScrollRequest.Builder()//
-						.scroll(new Time.Builder().time(TimeValue.timeValueMinutes(1).getMillis()
-								+ "ms").build())//
-						.scrollId(scroll_id).build(), Map.class);
-				hits = scroll_response.hits().hits();
+				if ( hits.size() > 0 )
+				{
+					previous_id = hits.stream()//
+					 .map(ElasticSearchCommon::parseSearchResultsToOneSearchResultWithTypeing)//
+					 .toList().get(hits.size() - 1).readAsAtom(new FieldName("id"), null);
+
+				}
 
 			}
-			while ( hits.size() != 0 ); // Zero hits mark the end of the scroll and the
+			while ( ! hits.isEmpty() ); // Zero hits mark the end of the scroll and the
 										// while
 										// loop.
-			return true;
+				CloudExecutionEnvironment.getSimpleCurrent().getSimpleSearch().closePointInTime(pitId);
```

Comment:
You don't have to used CloudExecutionEnvironment and ISearch to access closePointInTime. It's available locally.

Reply:
removed.